### ISSUE TITLE
Should {path} in access logs show value for {>Caddy-Rewrite-Original-URI}

### ISSUE THREAD

**Issue by tobya** [mholt/caddy#1185]
This is an related to  #1153 assuming Pull Request #1184 is merged.

In an access log the placeholder  `{>Caddy-Rewrite-Original-URI}` can be used with a fastcgi site that does a rewrite to output the origional path requested.  {path} will output the path after redirection

eg.

with a caddyfile

```
localhost:1234 {
log /  access.log "{path} {>Caddy-Rewrite-Original-URI} "

rewrite / {
 r (.*)
 to /redir.php
}

 fastcgi / localhost:34345  php
 root  D:\webroot\phpcgi
}
```

for the url  http://localhost:1234/customers/department/sales/2233 the log will show

> /redir.php /customers/department/sales/2233 

So, there is an argument here for {path} in the caddy file to log the same value as {>Caddy-Rewrite-Original-URI} as really it is the requested path that is of interest in a log rather than the rewritten path.

Is it worth considering another log placeholder such as `{rewrite_path}` which would contain what `{path}` currently contains, ie. the path after rewriting, and have `{path}` output what `{>Caddy-Rewrite-Original-URI}` does?

**Comment by abiosoft** [comment]
This is my view as well. https://github.com/mholt/caddy/issues/1153#issuecomment-250827388.

**Comment by elcore** [comment]
@tobya @abiosoft like this?

```
diff --git a/caddyhttp/httpserver/replacer.go b/caddyhttp/httpserver/replacer.go
index 02ad54f..bcaa2d8 100644
--- a/caddyhttp/httpserver/replacer.go
+++ b/caddyhttp/httpserver/replacer.go
@@ -221,8 +221,12 @@ func (r *replacer) getSubstitution(key string) string {
 		}
 		return host
 	case "{path}":
-		return r.request.URL.Path
+		return r.request.Header.Get("Caddy-Rewrite-Original-URI")
 	case "{path_escaped}":
+		return url.QueryEscape(r.request.Header.Get("Caddy-Rewrite-Original-URI"))
+	case "{rewrite_path}":
+		return r.request.URL.Path
+	case "{rewrite_path_escaped}":
 		return url.QueryEscape(r.request.URL.Path)
 	case "{query}":
 		return r.request.URL.RawQuery
diff --git a/caddyhttp/redirect/redirect_test.go b/caddyhttp/redirect/redirect_test.go
index 27998ab..a71cb3a 100644
--- a/caddyhttp/redirect/redirect_test.go
+++ b/caddyhttp/redirect/redirect_test.go
@@ -108,7 +108,7 @@ func TestParametersRedirect(t *testing.T) {
 
 	re = Redirect{
 		Rules: []Rule{
-			{FromPath: "/", Meta: false, To: "http://example.com/a{path}?b=c&{query}", RequestMatcher: httpserver.IfMatcher{}},
+			{FromPath: "/", Meta: false, To: "http://example.com/a{rewrite_path}?b=c&{query}", RequestMatcher: httpserver.IfMatcher{}},
 		},
 	}
 
diff --git a/caddyhttp/rewrite/rewrite_test.go b/caddyhttp/rewrite/rewrite_test.go
index 7468a19..f6bc291 100644
--- a/caddyhttp/rewrite/rewrite_test.go
+++ b/caddyhttp/rewrite/rewrite_test.go
@@ -24,7 +24,7 @@ func TestRewrite(t *testing.T) {
 	regexps := [][]string{
 		{"/reg/", ".*", "/to", ""},
 		{"/r/", "[a-z]+", "/toaz", "!.html|"},
-		{"/url/", "a([a-z0-9]*)s([A-Z]{2})", "/to/{path}", ""},
+		{"/url/", "a([a-z0-9]*)s([A-Z]{2})", "/to/{rewrite_path}", ""},
 		{"/ab/", "ab", "/ab?{query}", ".txt|"},
 		{"/ab/", "ab", "/ab?type=html&{query}", ".html|"},
 		{"/abc/", "ab", "/abc/{file}", ".html|"},
diff --git a/caddyhttp/rewrite/to_test.go b/caddyhttp/rewrite/to_test.go
index 5100a5f..5c01da1 100644
--- a/caddyhttp/rewrite/to_test.go
+++ b/caddyhttp/rewrite/to_test.go
@@ -22,7 +22,7 @@ func TestTo(t *testing.T) {
 		{"/?a=b", "/testfile /index.php?{query}", "/testfile?a=b"},
 		{"/?a=b", "/testdir /index.php?{query}", "/index.php?a=b"},
 		{"/?a=b", "/testdir/ /index.php?{query}", "/testdir/?a=b"},
-		{"/test?url=http://", " /p/{path}?{query}", "/p/test?url=http://"},
+		{"/test?url=http://", " /p/{rewrite_path}?{query}", "/p/test?url=http://"},
 		{"/test/?url=http://", " /{uri}", "/test/?url=http://"},
 	}
```

**Comment by tobya** [comment]
@elcore that would do the trick I guess.  Is there a worry here about backward compatibility or is that not a big concern at this time?

**Comment by elcore** [comment]
@tobya I guess it is okay to break backwards compatibility before 1.X.X, am I right @mholt?

**Comment by tw4452852** [comment]
Maybe we could make a snapshot of the request in `Logger.ServeHTTP` before going through the next middleware.

**Comment by elcore** [comment]
>Maybe we could make a snapshot of the request in `Logger.ServeHTTP` before going through the next middleware.

I am not sure how we could implement this (I would need to read through the code again -- I have vacation and I don't really want to do it), but I believe that it adds unnecessary complexity. -- We should use `Caddy-Rewrite-Original-URI` as it is pretty simple 😄

**Comment by mholt** [comment]
I wonder if we should start using context.Context for stuff like this.

**Comment by tobya** [comment]
@elcore would your fix as shown work if no redirection took place?

**Comment by elcore** [comment]
@tobya I am sorry, but no 😢 -- I don't have time to work on this particular issue (issue priority) (Reason: School Exams)

**Comment by mholt** [comment]
I think by the time Go 1.8 is released, we can switch to using `context.Context` in the http.Request and eliminate the need for the extra (hacky) header field.

Either way, I think @tobya graciously fixed this in 94af37087be04f90f66899b16687611c82bc753b.

**Comment by tobya** [comment]
Now that I am running 0.9.4 - this does need to be fixed.  We should ideally replace {path} with {>Caddy-Rewrite-Original-URI} and create a new placeholder {rewrite_path} to show what the url was rewritten to.  I will look at this when i get a chance.  I might use the code that @elcore  provided as a starting point.
